### PR TITLE
fix duplicate balance in genesis

### PR DESCRIPTION
**Pull request by pepoviola**
Fix balance duplication in genesis.

### CODE REVIEW

**Review by wirednkod — APPROVED** [review]

File: javascript/packages/orchestrator/src/chainSpec.ts [hunk 1]
```diff
@@ -119,6 +132,8 @@ export async function addBalances(specPath: string, nodes: Node[]) {
       }
     }
 
+    runtimeConfig.balances.balances = Object.entries(balanceMap);
```
Comment (l0r1s) [inline javascript/packages/orchestrator/src/chainSpec.ts:135]: Why do we have ```balances.balances``` here ?

File: javascript/packages/orchestrator/src/chainSpec.ts [hunk 2]
```diff
@@ -105,7 +117,8 @@ export async function addBalances(specPath: string, nodes: Node[]) {
             ? node.balance
             : stakingBond! + BigInt(1)
           : node.balance;
-        runtimeConfig.balances.balances.push([stash_key, balanceToAdd]);
+
+        balanceMap[stash_key] = balanceToAdd;
```
Comment (l0r1s) [inline javascript/packages/orchestrator/src/chainSpec.ts:121]:
```suggestion
        balanceMap[stashKey] = balanceToAdd;
```

File: javascript/packages/orchestrator/src/chainSpec.ts [hunk 3]
```diff
@@ -96,6 +96,18 @@ export async function addBalances(specPath: string, nodes: Node[]) {
...
+      ) => {
+        memo[balance[0]] = balance[1];
+        return memo;
+      },
+      {},
+    );
+
     for (const node of nodes) {
       if (node.balance) {
         const stash_key = node.accounts.sr_stash.address;
```
Comment (l0r1s) [inline javascript/packages/orchestrator/src/chainSpec.ts:113]:
```suggestion
        const stashKey = node.accounts.sr_stash.address;
```

File: javascript/packages/orchestrator/src/chainSpec.ts [hunk 4]
```diff
@@ -96,6 +96,18 @@ export async function addBalances(specPath: string, nodes: Node[]) {
   try {
     const chainSpec = readAndParseChainSpec(specPath);
     const runtimeConfig = getRuntimeConfig(chainSpec);
+    // Create a balance map
+    const balanceMap = runtimeConfig.balances.balances.reduce(
```
Comment (l0r1s) [inline javascript/packages/orchestrator/src/chainSpec.ts:100]: This ```balances.balances``` is strange to me, maybe we could find a better naming ? :thinking:

File: javascript/packages/orchestrator/src/chainSpec.ts (hunk 1 above)
Comment (wirednkod) [reply javascript/packages/orchestrator/src/chainSpec.ts:135]: This is how the chainspec is specified in [polkadot](https://github.com/paritytech/polkadot/blob/f963646940e3c38e359d1ecf76478d1fed36021c/node/service/src/chain_spec.rs#L324-L325) as far as I know

File: javascript/packages/orchestrator/src/chainSpec.ts (hunk 4 above)
Comment (wirednkod) [reply javascript/packages/orchestrator/src/chainSpec.ts:100]: This is not something we chose to name. Its coming from [Polkadot chainspec](https://github.com/paritytech/polkadot/blob/f963646940e3c38e359d1ecf76478d1fed36021c/node/service/src/chain_spec.rs#L324-L325)

Comment (pepoviola) [reply javascript/packages/orchestrator/src/chainSpec.ts:100]: Exactly, the naming here is from the chain-spec.

File: javascript/packages/orchestrator/src/chainSpec.ts (hunk 2 above)
Comment (pepoviola) [reply javascript/packages/orchestrator/src/chainSpec.ts]: Changed, thanks!

**Review by l0r1s — APPROVED** [review]

**Pull request merged**